talk/expert/cpp20spaceship.tex

```diff
@@ -16,216 +24,141 @@
+      \end{itemize}
     \end{itemize}
   \end{block}
 \end{frame}
 
 \begin{frame}[fragile]
-  \frametitlecpp[20]{The three-way comparison operator}
-  \scriptsize
-  \begin{exampleblock}{Example}
-    \begin{cppcode*}{}
-    struct Grade
-    {
-      double number ;
-      char letter ;
-      std::string name ;
-      auto operator<=>( Grade const & other ) const = default ;
-      std::ostream & operator<<( std::ostream & os, Grade const & g )
-      { return (os<<g.letter<<", "<<g.number<<", "<<g.name) ; }
-    } ;
-
-    int main()
-    {
-      std::set<Grade> grades
-      { { 19,'A',"Djamila"}, {12,'C',"Charles"}, {16.5,'A',"Marc"} } ;
-      for ( auto const & grade : grades )
-      { std::cout<<grade<<std::endl ; }
-    }
-    \end{cppcode*}
-  \end{exampleblock}
-  \begin{Verbatim}
-C, 12, Charles
-A, 16.5, Marc
-A, 19, Djamila
-  \end{Verbatim}
-\end{frame}
-
-\begin{frame}[fragile]
-  \frametitlecpp[20]{The three-way comparison operator}
-  \begin{block}{Default \texttt{<=>} implementation}
+  \frametitlecpp[20]{Compiler-generated \texttt{operator<=>}}
+  \begin{block}{Default \texttt{operator<=>} implementation}
     \begin{itemize}
-      \item We see above that the definition of \mintinline{cpp}{<=>} has been provided by the compiler, and the use of \mintinline{cpp}{<} by \mintinline{cpp}{std::set} to sort its elements has been rewritten by the compiler in terms of \mintinline{cpp}{<=>}.
-      \item If \mintinline{cpp}{<=>} is defaulted and no \mintinline{cpp}{==} is defined, then the compiler also provides a defaulted \mintinline{cpp}{==}.
+    \item One can ask the compiler to provide a default implementation for \mintinline{cpp}{operator<=>} and/or \mintinline{cpp}{operator==}
+      \begin{itemize}
+        \item declaring them with ``\mintinline{cpp}{= default}''
+        \item it will compare the member variables, one by one
+      \end{itemize}
+    \item Can be useful e.g. for tuples
+    \item Can be wrong, e.g. for the Ratio class
+    \item If \mintinline{cpp}{operator<=>} is defaulted and no \mintinline{cpp}{operator==} is defined, then the compiler also provides \mintinline{cpp}{operator==}
     \end{itemize}
   \end{block}
 \end{frame}
 
 \begin{frame}[fragile]
-  \frametitlecpp[20]{The three-way comparison operator}
+  \frametitlecpp[20]{\texttt{operator<=>} summary}
   \begin{alertblock}{Summary}
     \begin{itemize}
-      \item Defining \mintinline{cpp}{<=>} allows you to use \mintinline{cpp}{<}, \mintinline{cpp}{>},\mintinline{cpp}{<=}, and \mintinline{cpp}{>=} as well.
-      \item The standard library defines a few kinds of orderings (strong, weak and partial).
-      \item If \mintinline{cpp}{<=>} does not define a strong order, avoid to define and/or use \mintinline{cpp}{==}.
+      \item Defining \mintinline{cpp}{operator<=>} allows you to get \mintinline{cpp}{operator<}, \mintinline{cpp}{operator>},\mintinline{cpp}{operator<=}, and \mintinline{cpp}{operator>=} for free
```

Comment:
I originally deliberately suggested `allows you to use`, rather than `allow you to get`, so to not let think the reader that the operators are generated.

Reply:
indeed, you're fully right. Fixed